mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/zk/ZookeeperLeadershipFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.zk;
+
+import io.mantisrx.server.core.BaseService;
+import io.mantisrx.server.core.CoreConfiguration;
+import io.mantisrx.server.core.ILeaderElectorFactory;
+import io.mantisrx.server.core.ILeadershipManager;
+import io.mantisrx.server.core.json.DefaultObjectMapper;
+import io.mantisrx.server.core.master.ZookeeperLeaderMonitorFactory;
+import io.mantisrx.server.core.zookeeper.CuratorService;
+import io.mantisrx.server.master.config.ConfigurationProvider;
+import io.mantisrx.server.master.config.MasterConfiguration;
+import io.mantisrx.shaded.org.apache.curator.utils.ZKPaths;
+
+public class ZookeeperLeadershipFactory extends ZookeeperLeaderMonitorFactory implements ILeaderElectorFactory {
```

Comment:
This felt awkward to write, but was required since the interface is split and core needed a monitor factory.